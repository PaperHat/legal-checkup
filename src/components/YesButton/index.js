import React, { Fragment } from 'react'

import StyledYesButton from '../styled/Button/YesButton'

export default function YesButton ({ handleClick }) {
  return (
    <Fragment>
      <StyledYesButton onClick={handleClick}>Yes</StyledYesButton>
    </Fragment>
  )
}
