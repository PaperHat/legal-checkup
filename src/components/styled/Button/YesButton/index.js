import Button from '..'

const YesButton = Button.extend`
  border-right: 0px;
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
`

YesButton.displayName = 'YesButton'

export default YesButton
