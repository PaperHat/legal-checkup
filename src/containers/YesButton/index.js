import YesButton from '../../components/YesButton'
import { connect } from 'react-redux'
import { yesButtonClicked } from '../../state/actions'

function mapDispatchToProps (dispatch) {
  return {
    handleClick: () => dispatch(yesButtonClicked())
  }
}

const ConnectedYesButton = connect(null, mapDispatchToProps)(YesButton)
ConnectedYesButton.displayName = 'ConnectedYesButton'
export default ConnectedYesButton
