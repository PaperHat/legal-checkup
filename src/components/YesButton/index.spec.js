import React from 'react'
import { shallow } from 'enzyme'

import YesButton from '.'

describe('components:YesButton', function () {
  it('renders the YesButton and CSS properties properly', () =>
    expect(toJson(shallow(<YesButton>YesButton</YesButton>))).toMatchSnapshot())
})
