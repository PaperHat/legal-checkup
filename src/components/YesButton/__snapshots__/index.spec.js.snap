// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:YesButton renders the YesButton and CSS properties properly 1`] = `
<React.Fragment>
  <YesButton>
    Yes
  </YesButton>
</React.Fragment>
`;
